kasa/smart/smartchilddevice.py

```diff
@@ -28,17 +28,23 @@ def __init__(
         protocol: SmartProtocol | None = None,
     ) -> None:
         super().__init__(parent.host, config=parent.config, protocol=parent.protocol)
-        self._parent = parent
+        self._parent: SmartDevice = parent
         self._update_internal_state(info)
         self._components = component_info
         self._id = info["device_id"]
         self.protocol = _ChildProtocolWrapper(self._id, parent.protocol)
 
     async def update(self, update_children: bool = True):
+        """Call the parent update method."""
+        await self._parent.update()
+
+    async def _update_modules(self):
         """Update child module info.
 
```

Comment:
I didn't follow the whole code paths on this, but this sounds potentially dangerous as it changes the behavior (if we were calling it somewhere) and may cause extraneous update()s?

Reply:
Removed